### PR TITLE
Separate extras and package name in requirements YAML

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Fix the following error by separating package name and extras:

```
Your branch is up to date with 'origin/master'.
> python dev/update_requirements.py --requirements-yaml-location requirements/gateway-requirements.yaml
Updated pydantic.max_major_version to 2
Traceback (most recent call last):
  File "dev/update_requirements.py", line 74, in <module>
    main()
  File "dev/update_requirements.py", line 64, in main
    latest_major_version = get_latest_major_version(pip_release)
  File "dev/update_requirements.py", line 16, in get_latest_major_version
    response.raise_for_status()
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://pypi.org/pypi/uvicorn%5Bstandard%5D/json
```

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: requirements/gateway-requirements.yaml [hunk 1]
```diff
@@ -11,7 +11,9 @@ fastapi:
   max_major_version: 0
 
 uvicorn:
-  pip_release: uvicorn[standard]
```
Comment (harupy) [inline requirements/gateway-requirements.yaml:14]: There is no package named `uvicorn[standard]`. Extras need to be separated

**Review by WeichenXu123 — APPROVED** [review]

**Review by serena-ruan — APPROVED** [review]
LGTM!

File: dev/generate_requirements.py [hunk 2]
```diff
@@ -84,6 +84,8 @@ def generate_requirements_txt_content(requirements_yaml):
         pip_release = package_entry["pip_release"]
         version_specs = []
 
+        extras = f"[{','.join(package_entry['extras'])}]" if package_entry.get("extras") else ""
```
Comment (harupy) [inline dev/generate_requirements.py:87]: /ai can this python code be simplified using assignment expression?

**Review by BenWilson2 — APPROVED** [review]
Great fix @harupy :)

**Pull request merged**